docs/rules/import-type-order.md

```diff
@@ -0,0 +1,33 @@
+# Enforces order of import statement types.
+
+## Rationale
+- Improves readability and organization by grouping naturally related items together.
+
+## Rule Details
+Enforce order of import statement types. Imports should be listed in the following order:
+
+* Externals
+* Absolute paths
+* Parent directories
+* Siblings
+
+The following are considered warnings
+```js
+import Foo from './foo'; // Should come after external and parent imports.
+import Baz = '../baz'; // Should come after external imports
+import BarModule from 'BarModule';
+```
+
+The following import order is valid:
+
+```js
+export interface TheThreeStooges {
+import BarModule from 'BarModule';
```

Comment:
Wow, I don't even know why I wrapped those imports in an interface. Probably pulled code from somewhere and forgot to get rid of the export.